framework/wazuh/core/cluster/master.py

```diff
@@ -56,7 +56,9 @@ def done_callback(self, future=None):
             Synchronization process result.
         """
         super().done_callback(future)
-        self.wazuh_common.sync_integrity_free = True
+        # Integrity task is only freed if master is not waiting for Extra valid files.
+        if not self.wazuh_common.extra_valid_requested:
+            self.wazuh_common.sync_integrity_free = True
```

Comment:
We need to add some auxiliary mechanisms to ensure that this variable is correctly updated (exception case).

Reply:
Done.